Add taints support

lnovara opened the pull request:
The aim of this PR is to add Kubernetes taints support to this installer, this way we'll be able to better isolate workloads between node pools.

This change is updating the interface we expose for Kubernetes cluster creation from the installer hence, if we all agree about this change, I'll do the same with `fury-eks-installer` and `fury-gke-installer`.

KFD docs PR: sighupio/kfd-docs#41

lnovara committed:
fixed terraform error since oauth2_permissions is a set

lnovara committed:
added kubernetes taints support

lnovara committed:
bumped azurerm provider version to 2.16

lnovara committed:
updated example

lnovara committed:
updated module README

lnovara committed:
fixed docs link in main README

angelbarrera92 commented:
![Clap](https://media.giphy.com/media/Jf4FBs6JZhvKU/giphy.gif)

Looks awesome.
As you said, we are altering the interface to add a new attribute, so we have to include it as soon as possible (even without the logic) to EKS and GKE installers to match the new interface.

Thanks a lot Luca

lnovara commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Taints in Cloud Installers](https://trello.com/c/nTV78dTI/188-taints-in-cloud-installers)

lnovara commented:
I've updated the interface and added taints logic to the EKS and GKE installer as well:
- sighupio/fury-eks-installer#1
- sighupio/fury-gke-installer#2
